api/v1alpha1/clusterconfig_types.go

```diff
@@ -312,3 +317,42 @@ func (NFD) VariableSchema() clusterv1.VariableSchema {
 		},
 	}
 }
+
+type CSIProviders struct {
+	// +optional
+	Providers []*CSIProvider `json:"providers,omitempty"`
```

Comment:
Let me know if Im missing the reason for a pointer please.
```suggestion
	Providers []CSIProvider `json:"providers,omitempty"`
```